pio-proc/src/lib.rs

```diff
@@ -279,7 +284,12 @@ pub fn pio_file(item: TokenStream) -> TokenStream {
         Err(e) => return parse_error(e, &args.program).into(),
     };
 
-    to_codegen(program, args.max_program_size).into()
+    to_codegen(
+        program,
+        args.max_program_size,
+        args.file_path.into_os_string().into_string().ok(),
```

Comment:
Is it really `ok()` that we want? An error would be ignored and yield surprising results

Reply:
My reasoning for using `ok()` was that I thought the existing implementation of the `pio_file` macro would work with non-UTF-8 filenames. I didn't want to break that, but there's no way to pass a `Path` or `OsString` that's not also a valid `String` to `include_bytes`.

However I now tested this, and noticed that my assumption was wrong: `pio_file` only works if it gets passed a literal string, not (eg.) a byte-string (`b"...."`).

BTW, the current behavior if the first argument to `pio_file!` is not a string literal is also quite surprising: The parameter is silently ignored, and an empty program is parsed.

Anyway, it currently doesn't matter if we call `.ok()` or `.unwrap()`, the code in question is only called with paths generated from valid strings.

Would you prefer this change to make the code more robust against future changes?

```suggestion
        Some(args.file_path.into_os_string().into_string().expect("file path must be valid UTF-8")),
```